oci/spec_opts_windows.go

```diff
@@ -100,3 +100,32 @@ func escapeAndCombineArgs(args []string) string {
 	}
 	return strings.Join(escaped, " ")
 }
+
+// WithWindowsDevice adds a device exposed to a Windows (WCOW or LCOW) Container
+func WithWindowsDevice(IDType, ID string) SpecOpts {
+	return func(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
+		if len(IDType) == 0 {
+			return errors.New("missing IDType")
+		}
+		if s.Windows == nil {
+			s.Windows = &specs.Windows{}
+		}
+		s.Windows.Devices = append(s.Windows.Devices, specs.WindowsDevice{IDType: IDType, ID: ID})
+		return nil
+	}
+}
+
+// WithLCOWUVMDevice adds a device exposed from the LCOW kernel to an LCOW Container
+func WithLCOWUVMDevice(path string) SpecOpts {
+	return func(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
+		if len(path) == 0 {
+			return errors.New("missing path")
+		}
+		if s.Linux == nil {
+			return errors.New("spec is not LCOW")
+		}
```

Comment:
@TBBle For your question regarding this: I feel like it should be the caller's responsibility to use the correct `WithX` function. We could add a comment though about the implications of calling this function (that it "turns a WCOW spec into an LCOW spec") to make it clearer if you think that would suffice.